lib/segment/analytics/client.rb

```diff
@@ -310,9 +312,23 @@ def enqueue(action)
           ensure_worker_running
           @queue << action
         end
+        queue_logging(queue_full)
         !queue_full
       end
 
+      # private: Log warns and error when queue is getting full
+      def queue_logging(is_full)
+        queue_fullness_percentage = @queue.length.to_f / @max_queue_size
+        if queue_fullness_percentage >= 0.7 && queue_fullness_percentage < 1.0
+          logger.warn("Queue is #{queue_fullness_percentage * 100}% full.")
+        end
+        if is_full
+          msg = 'Queue is full, dropping events. ' \
+                'Please supply :max_queue_size value when initializing the client.'
+          logger.error(msg)
```

Comment:
I wonder if this should just be a warn, not error (it is recoverable after all).